bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -751,21 +753,17 @@ public boolean accept(long journalId) {
         }
     }
 
-    public void logAddEntry(ByteBuffer entry, WriteCallback cb, Object ctx) {
-        logAddEntry(Unpooled.wrappedBuffer(entry), cb, ctx);
-    }
-
     /**
      * record an add entry operation in journal.
      */
-    public void logAddEntry(ByteBuf entry, WriteCallback cb, Object ctx) {
+    public void logAddEntry(ByteBuf entry, boolean requiresForce, WriteCallback cb, Object ctx) {
```

Comment:
I think `ackAfterForce` is better than `requireForce`.

Reply:
ackAfterForce isn't clear either. we want to convey that we want to ack every time, and force some of the time. 
I think forceBeforeAck is better.